spec/features/publish/courses/new_engineers_teach_physics_spec.rb

```diff
@@ -29,6 +29,39 @@
     then_i_am_met_with_the_modern_languages_page
   end
 
+  scenario "changing from physics to another course clears the campaign_name params" do
+    when_i_visit_the_new_course_subject_page(:secondary)
+    and_i_select_a_subject(:physics)
+    and_i_click_continue
+    then_i_am_met_with_the_engineers_teach_physics_page(:secondary, :physics)
+    and_i_click_continue
+    then_i_see_an_error_message
+    and_i_select_an_option
+    and_i_click_continue
+    then_i_am_met_with_the_age_range_page(:secondary, :physics)
+
+    when_i_go_back
+    when_i_go_back
+    and_i_select_a_subject(:physics)
+    and_i_click_continue
+    then_i_am_met_with_the_engineers_teach_physics_page_with_etp(:secondary, :physics)
+    and_i_click_continue
+    then_i_am_met_with_the_age_range_page(:secondary, :physics)
+
+    when_i_go_back
+    when_i_go_back
+    and_i_select_a_subject(:latin)
+    and_i_click_continue
+    then_i_am_met_with_the_age_range_page_with_latin(:secondary, :latin)
+
+    when_i_go_back
+    and_i_select_a_subject(:physics)
+    and_i_click_continue
+    then_i_am_met_with_the_engineers_teach_physics_page_with_no_etp(:secondary, :physics)
+    and_i_click_continue
+    then_i_see_an_error_message
+  end
```

Comment:
Do we need this test if we have the other one? Is it testing anything that the other one isn't? It's a big feature test and our test suite is already quite slow!

Reply:
The new path validates in the controller, the edit path validates in the form_object, so I think we should cover both